### ISSUE TITLE
Types: non-existing fields are allowed

### ISSUE THREAD

**Issue by dodas** [graphql-editor/graphql-zeus#160]
Hey guys! Really great and unique project, keep it up!

I just noticed that when building query (either with Zeus or Chain), type definition allows for any field to be requested. This causes that the code builds find but errors at runtime, as GraphQL server will error when requesting non-existing fields.
For example:
```ts
const data = await chain.query({
  me: {
    email: true,
    name: true,
    // the "blablablabla" field doesn't exist on this type, but TS will not error
    blablablabla: true,
  },
});
```
Can we tighten the types so only actual fields from GraphQL are allowed?

Thanks!

**Comment by benny-medflyt** [comment]
This library: [graphql-typed-client](https://github.com/helios1138/graphql-typed-client) seems to have solved this issue. It might be worth looking at their code to see how they do it.

![tmp_23423623543](https://user-images.githubusercontent.com/24827547/107679162-3830fd80-6c6a-11eb-9df4-e12fc1865b1d.png)

**Comment by benny-medflyt** [comment]
Can you please confirm that this has been fixed? I've tried `graphql-zeus-3.0.5` and the issue still seems to be present.
Thank you

**Comment by IceBlizz6** [comment]
I got the same result, issue is still present with version 4.0.1
Is this a duplicate of #232 ?

I mentioned in my post there that issue only appears when an existing field has already been included in the query.
The person who looked at this bug may have closed it without testing it more extensively.
